.github/CONTRIBUTING.md

```diff
@@ -115,8 +115,35 @@ To get a report of current coverage, run the following:
 make coverage
 ```
 
+### Working with Optional Dependencies
+
+Langchain uses optional dependencies heavily to keep the package light.
+
+If you're adding a new dependency to Langchain, assume that it will be an optional dependency, and
+that most users won't have it installed.
+
+Users that do not have the dependency installed should be able to **import** your code without
+any side-effects (no warnings, no errors, no exceptions). 
+
+To introduce the dependency to the toml file, please do the following: 
+
+1. Add the dependency to the main group as an optional dependency
+  ```bash
+  poetry add --optional [package_name]
+  ```
+2. Add the dependency to the `extended_testing` extra (inside the pyproject.toml file)
+3. Update the poetry file
+  ```bash
+  poetry update
```

Comment:
should contributors be updating or should that be left to maintainers (avg pr could use `poetry lock --no-update` and we could intentionally update separately)?

Reply:
A few thoughts:

1. I haven't been able to get the extras to update with `poetry lock --no-update`
2. As part of testing we now do check the lock file to be consistent with the toml file -- so I'm not so worried about accepting updates from contributors anymore.

There might be an alternative -- I'm checking right now if we could specify an optional dep in the main group and same dep as a required one in another group (instead of in an extra)

Reply:
huh no-update has been working for me. 

im just wondering if updating our dependency versions (when it's not strictly needed) has potential to cause dependency conflicts for existing users (if their systems assume older version of the dependency). but if that's not a real concern then im happy with poetry update

Reply:
* it probably won't dependency conflicts since this is the lock file and isn't distributed to anyone
* But having to update the lock file creates merge conflicts which is a bit annoying.

* Confirming the `no-update` now

Reply:
Great catch @dev2049 -- looks like i was doing something wrong before

```sh
(test_actions) ➜  test_actions git:(main) ✗ cat poetry.lock | grep "cute"    
cute = []
(test_actions) ➜  test_actions git:(main) ✗ poetry lock --no-update      
Resolving dependencies... (0.1s)

Writing lock file
(test_actions) ➜  test_actions git:(main) ✗ cat poetry.lock | grep "cute"
cute = ["funcy"]
```